javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -872,8 +883,22 @@ public int lastIndexOf(T element, int end) {
     public <U> Vector<U> map(Function<? super T, ? extends U> mapper) {
         Objects.requireNonNull(mapper, "mapper is null");
 
-        final Iterator<? extends U> results = iterator().map(mapper);
-        return ofAll(results);
+        final MutableWrapper<Object[]> newLeading = new MutableWrapper<>(Arrays.empty());
```

Comment:
MutableWrapper is unnecessary overhead. Instead we should remove the `final` keyword and delete MutableWrapper.